lessons/projects/snake/snake_game.py

```diff
@@ -0,0 +1,143 @@
+import random
+from pathlib import Path
+
+import pyglet
+
+TILE_SIZE = 64
+TILES_DIRECTORY = Path('static/snake-tiles')
+
+class State:
+    def __init__(self):
+        self.snake = [(0, 0), (1, 0)]
+        self.snake_direction = 0, 1
+        self.width = 10
+        self.height = 10
+        self.food = []
+        self.add_food()
+        self.add_food()
+        self.snake_alive = True
+        self.queued_directions = []
+
+    def move(self):
+        if self.queued_directions:
+            new_direction = self.queued_directions[0]
+            del self.queued_directions[0]
+            old_x, old_y = self.snake_direction
+            new_x, new_y = new_direction
+            if (old_x, old_y) != (-new_x, -new_y):
+                self.snake_direction = new_direction
+
+        if not self.snake_alive:
+            return
+
+        old_x, old_y = self.snake[-1]
+        dir_x, dir_y = self.snake_direction
+        new_x = old_x + dir_x
+        new_y = old_y + dir_y
+
+        # Kontrola vylezení z hrací plochy
+        if new_x < 0:
+            self.snake_alive = False
+        if new_y < 0:
+            self.snake_alive = False
+        if new_x >= self.width:
+            self.snake_alive = False
+        if new_y >= self.height:
+            self.snake_alive = False
```

Comment:
Možná jestli by nebylo jednodušší a i zajímavější, kdyby hrací plocha byla cyklická? Tj. vylezu vpravo, přilezu zleva :)

Reply:
Logika je pak jednodušší, ale o hodně to zesložití vybírání správných obrázků pro části hada.

Reply:
@messa Je to jedna z možností rozšíření. Postupujeme tak, jak se vyvíjel had na Nokiích a vycházíme z hada z domácích projektů, kde také byla plocha ohraničená.